agent/agent.go

```diff
@@ -373,8 +371,8 @@ func (a *Agent) runTaskTimeoutCommands(ctx context.Context, tc *taskContext) {
 }
 
 // finishTask sends the returned EndTaskResponse and error
-func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string) (*apimodels.EndTaskResponse, error) {
-	err := a.uploadToS3(ctx, tc)
+func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string) (resp *apimodels.EndTaskResponse, err error) {
```

Comment:
In general we avoid named return values in Evergreen code. It tends to be more readable to declare variables close to where they are used. To avoid re-declaring `err` on https://github.com/evergreen-ci/evergreen/pull/2082/files#diff-6ea70e0a4de93ebcb83000a028edc163R403, you can write `var resp *apimodels.EndTaskResponse` immediately before that line.